nipoppy/layout.py

```diff
@@ -126,7 +136,11 @@ def path_labels(self) -> list[str]:
     @cached_property
     def path_infos(self) -> list[PathInfo]:
         """Return a list of all PathInfo objects defined in the layout."""
-        return [getattr(self, path_label) for path_label in self.path_labels]
+        return [
+            getattr(self, path_label)
+            for path_label in self.path_labels
+            if getattr(self, path_label) is not None
```

Comment:
your battle below depends on the meaning of "defined" in the docstring. If I am to follow the current spec:
>         """Return a list of all PathInfo objects defined in the layout."""

the code should be

```suggestion
            if isinstance(getattr(self, path_label), PathInfo)
```

if you mean

>        """Return a list of all PathInfo objects for all non-degenerate paths in the layout."""

then may be smth like

```suggestion
            if getattr(self, path_label)
```

and so on... what is "defined"?